src/routes/preHandlers/courses.js

```diff
@@ -115,12 +115,19 @@ exports.checkInterlocutors = async (req, courseCompanyId) => {
 exports.authorizeCourseEdit = async (req) => {
   try {
     const { credentials } = req.auth;
-    const { course } = req.pre;
-    if (course.archivedAt) throw Boom.forbidden();
+    const course = await Course
+      .findOne({ _id: req.params._id })
+      .populate({ path: 'slots', select: 'startDate endDate' })
+      .populate({ path: 'trainees', select: 'contact.phone', populate: { path: 'company' } })
+      .populate({ path: 'slotsToPlan' })
+      .populate({ path: 'contact' })
+      .lean();
+    if (!course || course.archivedAt) throw Boom.forbidden();
```

Comment:
Il me semble qu'on fait la distinction habituellement entre not found et forbidden

Reply:
Oui tu as raison, c'est modifié